src/cmd/ksh93/sh/jobs.c

```diff
@@ -598,40 +597,17 @@ void job_init(Shell_t *shp, int lflag) {
 #endif /* NTTYDISC */
 
     job.mypgid = getpgrp();
-    // Some systems have job control, but not initialized.
-    if (job.mypgid <= 0) {
```

Comment:
I am not sure about other systems, but on Linux `getpgrp()` should return 0 only if ksh is started as first process i.e. pid 1. Given that is it safe to remove this code @msekletar ?

Reply:
Pid 1 has process group 1 on Linux, so my previous comment was incorrect.